indico/modules/auth/controllers.py

```diff
@@ -584,6 +585,9 @@ def validate_everything(self, data, **kwargs):
                     raise ValidationError(_('This field cannot be empty.'), 'first_name')
                 if 'last_name' not in data['synced_fields'] and not data['last_name']:
                     raise ValidationError(_('This field cannot be empty.'), 'last_name')
+                for field in data['synced_fields']:
+                    if field not in multipass.locked_fields:
+                        raise ValidationError('Locked fields must be synced.', field)
```

Comment:
Careful, this need to take the edge case into account where there is no data for a synced field. In that case it should NOT be considered locked (otherwise such a user could not sign up).

Reply:
I think adding a ` and self.identity_info['data'].get(field)` will fix that but is there any way of easily testing this?

Reply:
yeah, hacking your local multipass-cern or multipass-jacow code to send empty data ;)

Reply:
also, maybe should we only allow desyncing mandatory fields (when empty)?

Reply:
not sure what you mean - as an admin you should always be able to desync (and in that case IMHO the user should be able to freely change the field), regardless if it's mandatory/empty or not

Reply:
im asking if we should let normal users desync empty fields if they are not mandatory
since this code is only for the signup page I don't think we even need to take admins into account...

Reply:
i assumed only the mandatory fields should be desynced. just pushed some code that seems to work well